### PR TITLE
refactor(framework) Add `try`-`except` block to `flwr stop` for `JSON` output

### PR DESCRIPTION
**Pull request by chongshenng**

### CODE REVIEW

**Review by jafermarq — APPROVED** [review]

File: src/py/flwr/cli/stop.py [hunk 1]
```diff
@@ -48,34 +48,39 @@ def stop(
...
+                bold=True,
+            )
+            raise typer.Exit(code=1) from err
+        finally:
+            channel.close()
+    except (typer.Exit, Exception):  # pylint: disable=broad-except, W0612  # noqa: F841
+        pass
     finally:
-        channel.close()
+        pass
```
Comment (danieljanes) [inline src/py/flwr/cli/stop.py:83]: We can remove `finally` if we don't use it

File: src/py/flwr/cli/stop.py [hunk 2]
```diff
@@ -48,34 +48,39 @@ def stop(
...
+            typer.secho(
+                f"❌ {err}",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1) from err
+        finally:
+            channel.close()
+    except (typer.Exit, Exception):  # pylint: disable=broad-except, W0612  # noqa: F841
+        pass
```
Comment (danieljanes) [inline src/py/flwr/cli/stop.py:81]: This "swallows" the exception, which mean `typer.Exit` ceases to work

File: src/py/flwr/cli/stop.py (hunk 1 above)
Comment (chongshenng) [reply src/py/flwr/cli/stop.py:83]: We are using it in #4710. It's not very ideal, but the diff is reduced.

File: src/py/flwr/cli/stop.py (hunk 2 above)
Comment (chongshenng) [reply src/py/flwr/cli/stop.py:81]: Similar to above, we will catch the `typer.Exit` in #4710.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**